### PR TITLE
spglib: 2.0.2 -> 2.1.0

### PR DESCRIPTION
**Pull request by markuskowa**
## Description of changes

Changelog: https://github.com/spglib/spglib/releases/tag/v2.1.0

@PsyanticY Please have look at `python3Packages.spglib`. I tried to update it as well, but there are a lot of changes that require some more work.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by markuskowa** [comment]
@GrahamcOfBorg build cp2k python3Packages.spglib

**Review by sheepforce — APPROVED** [review]

**Review by Janik-Haag — COMMENTED** [review]
LGTM; except for the nitpick.

File: pkgs/development/libraries/spglib/default.nix [hunk 1]
```diff
@@ -1,20 +1,22 @@
...
   pname = "spglib";
-  version = "2.0.2"; # N.B: if you change this, please update: pythonPackages.spglib
+  version = "2.1.0"; # N.B: if you change this, please update: pythonPackages.spglib
 
   src = fetchFromGitHub {
     owner = "spglib";
     repo = "spglib";
     rev = "v${version}";
-    sha256 = "sha256-8Voepj35CMbboL3Dc55Gc4+OLPTTSgqVQuvNcRQsqmU=";
+    sha256 = "sha256-EL3jkzyurc8fnzk9kAdTaEtLfLlLtmaVDFwChfCDOrQ=";
```
[the remark this answers is not in the document]
Comment (markuskowa) [reply pkgs/development/libraries/spglib/default.nix]: `grep -R 'sha256 = "sha256' pkgs | wc -l` yields 7863 results. This should simply be done by a search-and-replace script.

[the remark this answers is not in the document]
Comment (Janik-Haag) [reply pkgs/development/libraries/spglib/default.nix]: to be honest in my opinion we should have a linter for stuff like that but :shrug:
we'll probably get a formatter in a few weeks so thats a start ^^

**Review by Janik-Haag — APPROVED** [review]
LGTM

**Pull request merged**